### DISCUSSION TITLE
HL7 ERR segements processed as ACKS it seems

### DISCUSSION THREAD

**Discussion by azdave2020** [nextgenhealthcare/connect#5628] [Q&A] [answered]
What exactly is an ERR message?  
I have a vendor that seems to use it two different ways. I thought an ERR was an error??
Example 1 message rejected 
ERR||RXA^1^11^1^4|100^Segment sequence error^HL70357|E||||vaccination facility id is missing - Message Rejected|

example 2, they accepted (I unidentified it)
ERR|||0^Message accepted^HL70357|I||||Patient xxx"name" with 1 vaccination accepted into vaccination staging table|

Dave

**Comment by pacmano1** [comment]
ERR may in fact be informational in the IIS/HL7 space.   But always good to ask the other end.

**Comment by jonbartels** [comment, marked as answer]
https://hl7-definition.caristix.com/v2/HL7v2.6/Segments/ERR which cites the standard table for ERR.4 https://hl7-definition.caristix.com/v2/HL7v2.6/Tables/0516

In the samples provided the ERR.4 is E and I. So the second example is denoted as an informational message. 

Are the ACK codes for each message correctly indicating success or failure? AE AA and so on?

**Comment by azdave2020** [comment]
Thanks, fantastic info, So the root problem is that I get these back from the state into Mirth.
I can't figure out how to pass that AE info back to the EHR. I tried all the different responses in the Source tab, but I can't get it to give back.
Alternatively, how about I get Mirth to flag it as an error and show it on the dashboard? This data is unidentified 

MSH|^~\&|ASIIS^^|ASIIS^^|EPIC^^|60021^^|20230121101344-0700||ACK^V04^ACK|6062912689.102512833|P|2.5.1|||NE|NE|||||Z44^CDCPHINVS^^| 
MSA|AE|300408| 
ERR||RXA^1^11^1^4|103^Table value not found^HL70357|E||||vaccination facility id is unrecognized administered| ERR||RXA^1^11^1^4|100^Segment sequence error^HL70357|E||||vaccination facility id is unrecognized administered - Message Rejected|

**Reply by pacmano1** [reply]
Not sure I understand but you can set the response data type to raw and manually serialzie the message and do whatever you want include setting the status to ERROR, SENT, etc.

**Reply by azdave2020** [reply]
Let me clarify. The AE comes into MIRTH from the State Registry.     I need Mirth to inform the EHR (Mirth is in the middle between the EHR and the State)  of this error.  I am not seeing  how to set the response to Epic EHR as RAW or what you mean by serializing. Please elaborate
![Snag_f1a6230](https://user-images.githubusercontent.com/90641786/214165378-57aa32c5-67e8-4581-955f-fab7273b7dd7.png)
![Snag_f1a7ff9](https://user-images.githubusercontent.com/90641786/214165404-d1f9ed10-a238-4436-9fe0-b1a366dfe7bd.png)

**Reply by pacmano1** [reply]
You are calling the IIS endpoint in one of your destinations I assume.   The _response transformer of that destination_ would be where you would set the data type to raw and serialize and extract the message parts as needed to then make a decision about what to return to the sender to the listener you show above.    That might mean you create a $r value in the response transformer that would then appear in that drop down you are showing.   

I think you are looking for the easy button, their ain't one.  IMHO, to properly do this, you need to let mirth always fire the response transformer so you can inspect the actual results of the call to the IIS.   In order for the response transformer to _always_ fire, it must be set to raw.    

Let's assume you call the IIS via an http sender (not a web service sender) and the response HL7 message looks like:

`{ "response" : "<hl7 message here>"}`

After setting the response transformer to raw:

1) You must check the response code (e.g. was it 200?).  If not a 200, do you want to retry X times and set responseStatus to error?
2) if a 200, you must JSON.parse the response (after all you told mirth it was RAW).
3) You would then typically serialize the HL7 message in the JSON blob and make decisions about that (e.g. was the ERR an error or just informational)
4) You might build a $r response var with what you want to send to the caller of your channel.

This is not exhaustive, and there are multiple ways to attack this.

**Reply by azdave2020** [reply]
Thanks that helps alot!
Dave
